buildSrc/genVSproperties.bat

```diff
@@ -21,79 +21,69 @@ REM Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
 REM or visit www.oracle.com if you need additional information or have any
 REM questions.
 
-REM Windows bat file that runs vcvars32.bat for Visual Studio 2003
-REM   and echos out a property file with the values of the environment
-REM   variables we want, e.g. PATH, INCLUDE, LIB, and LIBPATH.
+setlocal ENABLEDELAYEDEXPANSION
+
+REM Windows bat file that runs vcvars64.bat for Visual Studio
+REM and echos out a property file with the values of the environment
+REM variables we want, e.g. PATH, INCLUDE, LIB, and LIBPATH.
 
 REM Clean out the current settings
 set INCLUDE=
 set LIB=
 set LIBPATH=
 
-REM Run the vsvars32.bat (12.0) / vcvars32.bat (15.0) file, sending it's output to neverland.
-REM The current officially supported Visual Studio version is 15.0.
-REM Handling of 11.0 and 14.0 is excluded here.
-REM The previous officially supported VS version was 12.0
-REM So, the search order is 150, then 120, then 100
-set VSVER=150
-set "VSVARS32FILE=C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\VC\Auxiliary\Build\vcvars32.bat"
-if not "%VS150COMNTOOLS%"=="" (
-    set "VS150COMNTOOLS=%VS150COMNTOOLS%"
+REM The current officially supported Visual Studio version is VS 2022
+
+REM Try the following in order of priority:
+REM 1. The VSCOMNTOOLS env var
+REM 2. The legacy VS150COMNTOOLS env var
+REM 3. Look in standard locations for Visual Studio (2022,2019,2017)
+
+
+set AUXBUILD=VC\Auxiliary\Build
+
+if not "%VSCOMNTOOLS%"=="" (
+    set "VSTOOLSDIR=%VSCOMNTOOLS%"
+) else if not "%VS150COMNTOOLS%"=="" (
+    set "VSTOOLSDIR=%VS150COMNTOOLS%"
 ) else (
-  if exist "%VSVARS32FILE%" set "VS150COMNTOOLS=C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\VC\Auxiliary\Build"
+    for %%a in (2022, 2019, 2017) do (
+        set year=%%a
+        for %%b in (Enterprise, Professional, Community) do (
```

Comment:
Quick two cents from me on this part - Microsoft also offers a free CLI-only version of Visual Studio called "Build Tools for Visual Studio" - https://visualstudio.microsoft.com/downloads/#build-tools-for-visual-studio-2022

Those install in a separate "BuildTools" directory (so ex. `C:\Program Files\Microsoft Visual Studio\2022\BuildTools\...`). Internally the directory structure is the same and all the scripts we need are also there, so it seems like this should work. I think we could also add BuildTools as an option here? I can check on my machine if it builds as part of this review.

Reply:
Thanks, that would be helpful. Let me know what you find.

Reply:
I decided to go ahead and add this. Can you test and review it?